### DISCUSSION TITLE
월요일에 할 것 같긴 한데 기록 차원에서....

### DISCUSSION THREAD

**Discussion by jpham005** [GunGonGamLeeLee/ft_transcendence#39] [4. Random]
이제 sidebar의 친구 목록, 랭킹 목록은 구현을 했고, 채팅방에서 해당 방의 유저 리스트를 받아오는 로직도 api만 있으면 금방 구현 가능해 보입니다.
이에 따라 백엔드에서도
1. 내 정보
2. 전체 친구 리스트
3. 전체 랭킹 리스트
4. 채널 유저 리스트
5. 채널 방 리스트 ( [ {public + protected} , {참여중}, {dm}] ) 한번 요청으로 다 받으면 편하긴 할듯.
6. 채널 입장 비밀번호 인증 - ok / notok 만 주면 될 것 같습니다.

를 보여줄 수 있는 api를 확정지으면 좋을 것 같네용.

그리고 이 모든것은 frontend/src/mocks/handlers.ts 에 있습니다.
앞으로도 백엔드에 아직 없는 기능은 이쪽에 추가할 예정입니다.

**Comment by Arat5724** [comment]
6. 채널 입장 비밀번호 인증 - ok / notok 만 주면 될 것 같습니다. 
- 입장 전/입장 후의 인증 로직을 상의해야 할 것 같습니다.
아니면 벌써 다 생각해놓으셨나??

**Reply by mypark42** [reply]
.

**Comment by mypark42** [comment]
1. 내 정보가 구체적으로 어떤 것들이 필요한지,

2-4. 리스트에서 유저 정보가 어떤 것들이 필요한지,

적어주세요!

**Reply by mypark42** [reply]
현재 db 구조 입니다 #26

**Reply by mypark42** [reply]
예시
```
{
  "chid": 1,
  "chName": "42GunGonGamLeeLee",
  "chOwnerId": 0,
  "display": true,
  "isLocked": false,
  "password": "01234",
  "chOwner": {
    "uid": 0,
    "displayName": "jaham"
  }
}
```
